server/journeys/start/controller.ts

```diff
@@ -9,8 +10,21 @@ export class StartJourneyController {
     (url: string) =>
     async (req: Request, res: Response): Promise<void> => {
       const { journeyId, prisonerNumber } = req.params
-      const prisoner = await this.prisonerSearchService.getPrisonerDetails(req, prisonerNumber as string)
+      const prisoner = await this.prisonerSearchService
+        .getPrisonerDetails(req, prisonerNumber as string)
+        .catch((error: unknown) => {
+          if (!(error instanceof Error)) {
+            throw error
+          }
+          const sanitisedError = error as SanitisedError
+          if (sanitisedError.status === 404) {
+            res.redirect(`${res.locals.digitalPrisonServicesUrl}/prisoner-search`)
+            return
+          }
```

Comment:
do we need to restrict it to 404 error only?
what if it fails a regex test and get a 400 bad request error?

Reply:
It's a good point and something we need to decide for all our calls like this - the ticket spec was just on 404s, but worth thinking about this more broadly.

Reply:
redirecting back to DPS seems like a catch-all handling to me, hence i feel that it should cover all 4XX 5XX errors.
it's probably very rare for such edge cases to happen tho.

Reply:
It would be perhaps jarring for a user to be redirected back to the DPS home screen because of an error the server made (5xx) or that we made in our request/response/setup (400, 401, 403 etc), I think? Maybe it would be worth redirecting the user to a generic error page (we have a very basic one)?

Reply:
I agree with error page for non 404s. The 400 case, while a good suggestion, would only happen if the user entered a manual URL. For the 5XXs we'd want to show the user a friendly error page

Reply:
We actually already have a middleware for this - I've added an extra test to ensure it's going there when we 500 for `getPrisonerDetails`